Addition of scenarios for copying/cutting/pasting text and redo, as well as toggling on and off.

yanncloud opened the pull request:
I have written a test for the scenario of copying, cutting, and pasting text. It calls the test page, writes a sample text, copies it, deletes it, and then pastes it back. Then it writes another sample text, cuts it, and pastes it back.
After that, I wrote a test for the redo scenario. It calls the test page, adds an element, writes a sample text, clicks on Undo, and then on Redo.
Afterwards, I attempted to write the tests for the toggle on and off scenario. However, I encountered the problem that without IDs for the text menu, I am unable to verify the changes, such as text in italics or a hyperlink.

Entkenntnis commented:
Thank you @yanncloud for the tests! I'm very glad to see your first contribution!

Entkenntnis commented on tests/400-serlo-editor.ts:
```diff
@@ -89,3 +109,124 @@ Scenario('Markdown list shortcut', async ({ I }) => {
+
+  I.click('Some text')
+
+  I.amOnPage('https://serlo.org')
```
Can you maybe explain why you are leaving the editor (probably to reset content?)

You can use `I.amOnpage('/')` to make it independent of the domain

elbotho commented on tests/400-serlo-editor.ts:
```diff
@@ -89,3 +109,124 @@ Scenario('Markdown list shortcut', async ({ I }) => {
+
+  I.see('TESTTESTTEST')
+
+  for (let i = 0; i < 12; i++) {
+    I.pressKey('Backspace')
+  }
```
very tidy, but you could probably also just leave it there? :)

yanncloud commented:
> > Afterwards, I attempted to write the tests for the toggle on and off scenario. However, I encountered the problem that without IDs for the text menu, I am unable to verify the changes, such as text in italics or a hyperlink.
> 
> is this still the case?

I have implemented your tips, so now the functionality of bold/italic and the link is checked without the need for IDs. I have modified the link validation in a way that I don't have to leave the editor anymore. Additionally, I have already implemented the 'Toggle On' feature for math.

Entkenntnis reviewed (approved):
thank you - works nicely

yanncloud commented:
> @yanncloud four tests are failing for me. Are they working for you?
> - Copy/cut/paste text
> - Keyboard Toggle on and off
> - Unordered list shortcuts
> - Toolbar Toggle on and off

Yes they are working for me.

yanncloud commented:
Thank you. I add this to the tests. ☺️

elbotho reviewed (approved):
thank you @yanncloud
